src/compas/robots/model/joint.py

```diff
@@ -608,10 +613,28 @@ def __init__(self, name, type, parent, child, origin=None, axis=None, calibratio
         self.attr = kwargs
         self.child_link = None
         self.position = 0
-        # The following are world-relative frames representing the origin and the axis, which change with
-        # the joint state, while `origin` and `axis` above are parent-relative and static.
-        self.current_origin = self.origin.copy()
-        self.current_axis = self.axis.copy()
+
+    @property
+    def origin(self):
+        return self._origin
+
+    @property
+    def axis(self):
+        return self._axis
+
+    @origin.setter
+    def origin(self, value):
+        self._origin = value
+        # The following is a world-relative frame representing the origin, which change with
+        # the joint state, while `origin` is parent-relative and static.
+        self.current_origin = value.copy()
```

Comment:
maybe i haven't properly thought this through, but should this be 
`self.current_origin = self.current_origin @ self.origin.inverse() @ value`
where i'm ignoring the distinction between a frame and its transformation